model/CompatibilityCheckerService.php

```diff
@@ -23,12 +23,17 @@
 
 use oat\taoClientDiagnostic\model\browserDetector\WebBrowserService;
 use oat\taoClientDiagnostic\model\browserDetector\OSService;
+use oat\oatbox\service\ConfigurableService;
 
-
-class CompatibilityChecker
+class CompatibilityCheckerService extends ConfigurableService
```

Comment:
you should call the parent construct in case of options
CompatibilityFile could be one option of this service